### ISSUE TITLE
CKV_DOCKER_7 only reports on first occurence of issue

### ISSUE THREAD

**Issue by Kemosabert** [bridgecrewio/checkov#6890]
**Describe the issue**
The rule `CKV_DOCKER_7` only reports the first occurence of the rule in a file and does not flag on other stages also not pinning the image to a specific tag.

**Examples**
```Dockerfile
FROM test.dkr.ecr.eu-west-2.amazonaws.com/my-registry/baseimage as baseImage

# do bunch of things here

FROM test.dkr.ecr.eu-west-2.amazonaws.com/my-registry/migrations-runner:latest as setup
# do bunch of more things here

ENTRYPOINT ["/entrypoint.sh"]
```

I would expect it to flag both FROM statements as problematic, but it only flags the first one. Once that is fixed, the 2nd one will also be flagged.

**Version (please complete the following information):**
 - Checkov Version [3.2.327]

**Additional context**
